geoapps_utils/driver/data.py

```diff
@@ -163,3 +163,15 @@ def write_ui_json(self, path: Path) -> None:
         :param path: Path to write the ui.json file.
         """
         self.input_file.write_ui_json(path.name, str(path.parent))
+
+    def add_options(self):
+        """Set out_groups options with current state and data."""
+
+        dump = self.model_dump()
+        dump["geoh5"] = str(dump["geoh5"].h5file.resolve())
+        out_group = dump.pop("out_group")
```

Comment:
Need to check if `self` has an `out_group` and it is not None.  

Probably better if this method just returns a demoted/stringified dictionary only. The update to out_group can happen on the sub-class directly depending on out_group.

Reply:
done